### PR TITLE
#28 user managment delete user command+handler

### PR DESCRIPTION
**Pull request by kevjo15**
# Pull Request Title
delete user command+handler
## Description

I made a DeleteUserByIdCommand and handler. also addes attribut boolen IsDeleted to UserModel

## Changes Made

change UserModel class. added Boolen IsDeleted

## Related Ticket or Issue

#49 

## Checklist

- [x] Code adheres to the coding conventions
- [] Unit tests are added for new code
- [] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [x] Code formatting complies with the `.editorconfig` file
- [x] Pull request template is filled out

### CODE REVIEW

File: Application-Layer/Queries/GetUserById/GetUserByIdQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+
+namespace Application_Layer.Queries.GetUserById
+{
+    public class GetUserByIdQueryHandler : IRequestHandler<GetUserByIdQuery, IdentityUser>
```
Comment (BojanMirkovic) [inline Application-Layer/Queries/GetUserById/GetUserByIdQueryHandler.cs:11]: same IdentityUser ?

File: Application-Layer/Commands/DeleteUser/DeleteUserCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    //        var user = await _userRepository.GetUserByIdAsync(request.UserId);
+    //        if (user == null)
+    //        {
+    //            throw new InvalidOperationException($"User with ID {request.UserId} was not found.");
+    //        }
+
+    //        var success = await _userRepository.DeleteUserByIdAsync(user);
+    //        if (!success)
+    //        {
+    //            throw new InvalidOperationException("Failed to update the user.");
```
Comment (BojanMirkovic) [inline Application-Layer/Commands/DeleteUser/DeleteUserCommandHandler.cs:34]: I think you want to send message "Failed to delete the user."

File: Application-Layer/Commands/DeleteUser/DeleteUserCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    //public DeleteUserCommandHandler(IUserRepository userRepository)
+    //{
+    //    _userRepository = userRepository;
+    //}
+
+    //public async Task<bool> Handle(DeleteUserCommand request, CancellationToken cancellationToken)
+    //{
+    //    try
+    //    {
+    //        var user = await _userRepository.GetUserByIdAsync(request.UserId);
```
Comment (HubTicaBB) [inline Application-Layer/Commands/DeleteUser/DeleteUserCommandHandler.cs:25]: You can inject `UserManager` from the Identity and use the existing `FindByIdAsync` function.

File: Domain-Layer/Models/UserModel/UserModel.cs [hunk 4]
```diff
@@ -7,5 +7,6 @@ public class UserModel : IdentityUser
         public string? FirstName { get; set; }
         public string? LastName { get; set; }
         public string? Role { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (HubTicaBB) [inline Domain-Layer/Models/UserModel/UserModel.cs:10]: Don't forget db migration

**Review by Nemanja1208 — APPROVED** [review]

**Pull request merged**